node/ChainwebNode.hs

```diff
@@ -161,6 +166,47 @@ runCutMonitor logger db = L.withLoggerLabel ("component", "cut-monitor") logger
             -- $ S.map _cutMap
             -- $ cutStream db
 
+
+runAmberdataBlockMonitor :: Logger logger => logger -> CutDb cas -> IO ()
+runAmberdataBlockMonitor logger db = L.withLoggerLabel ("component", "amberdata-block-monitor") logger $ \l -> do
+    go l `catchAllSynchronous` \e ->
```

Comment:
should this be wrapped into `runForever`. That function would take care of catching exceptions, long errors, and restart as needed.